jbpm-benchmarks/jbpm-performance-tests/src/main/java/org/jbpm/test/performance/jbpm/JBPMController.java

```diff
@@ -104,26 +99,33 @@ public void clear() {
     public void tearDown() {
         clear();
         if (persistence) {
-            if (emf != null) {
-                emf.close();
-                emf = null;
-                EntityManagerFactoryManager.get().clear();
-
-            }
-            if (ds != null) {
-                ds.close();
-                ds = null;
-            }
             try {
+                if (emf != null) {
+                    emf.close();
+                    emf = null;
+                    EntityManagerFactoryManager.get().clear();
+
+                }
+                if (ds != null) {
+                    ds.close();
+                    ds = null;
+                }
+
                 InitialContext context = new InitialContext();
-                UserTransaction ut = (UserTransaction) context.lookup(
-                        JtaTransactionManager.DEFAULT_USER_TRANSACTION_NAME);
+                UserTransaction ut = (UserTransaction) context.lookup(JtaTransactionManager.DEFAULT_USER_TRANSACTION_NAME);
                 if (ut.getStatus() != Status.STATUS_NO_TRANSACTION) {
                     ut.setRollbackOnly();
                     ut.rollback();
                 }
             } catch (Exception e) {
-                // do nothing
+                e.printStackTrace();
+            } finally {
+                // Terminate Narayana Recovery Manager
+                try {
+                    RecoveryManager.manager().terminate();
+                } catch (IllegalStateException e) {
+                    log.info("RecoveryManager is already closed.");
```

Comment:
Is it an info message? Warning would be more appropriate in this case.

It would be good to include the exception too - it's just your assumption, that the "RecoveryManager is already closed."

Reply:
@MarianMacik that's right, on the other hand, in theory, this contract can change. How about keeping the info as is and logging the entire exception on debug level?